src/Razor/src/Microsoft.NET.Sdk.Razor/ApplicationPartsProvider.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class ApplicationPartsProvider
+    {
+        private readonly HashSet<string> _mvcAssemblies;
```

Comment:
Worth reviewing this class? Or is this all as-is from MVC?

Reply:
It'd be worth looking at. It's loosely based on the MVC one, but the lookup of dependencies uses MetadataReader and has different assumptions than deps.